testproject/Assets/Tests/Manual/SceneTransitioningAdditive/DespawnInSceneNetworkObject.cs

```diff
@@ -8,87 +8,91 @@ namespace TestProject.ManualTests
     /// <summary>
     /// Used for manually testing spawning and despawning in-scene
     /// placed NetworkObjects
-    ///
-    /// Note: We do not destroy in-scene placed NetworkObjects, but
-    /// users must handle visibility (rendering wise) when the in-scene
-    /// NetworkObject is spawned and despawned.  This class just enables
-    /// or disabled the mesh renderer.
     /// </summary>
     public class DespawnInSceneNetworkObject : NetworkBehaviour
     {
+        [Tooltip("When set, the server will despawn the NetworkObject upon its first spawn.")]
+        public bool StartDespawned;
+
         private Coroutine m_ScanInputHandle;
-        private MeshRenderer m_MeshRenderer;
 
-        private void Start()
-        {
-            if (!IsSpawned)
-            {
-                m_MeshRenderer = GetComponent<MeshRenderer>();
-                if (m_MeshRenderer != null)
-                {
-                    m_MeshRenderer.enabled = false;
-                }
-            }
-        }
+        // Used to prevent the server from despawning
+        // the in-scene placed NetworkObject after the
+        // first spawn (only if StartDespawned is true)
+        private bool m_ServerDespawnedOnFirstSpawn;
+
+        private NetworkManager m_CachedNetworkManager;
 
         public override void OnNetworkSpawn()
         {
             Debug.Log($"{name} spawned!");
-            m_MeshRenderer = GetComponent<MeshRenderer>();
-            if (m_MeshRenderer != null)
-            {
-                m_MeshRenderer.enabled = true;
-            }
+
             if (!IsServer)
             {
                 return;
             }
+
+            m_CachedNetworkManager = NetworkManager;
+
             if (m_ScanInputHandle == null)
             {
-                m_ScanInputHandle = StartCoroutine(ScanInput());
+                // Using the NetworkManager to create the coroutine so it is not deactivated
+                // when the GameObject this NetworkBehaviour is attached to is disabled.
+                m_ScanInputHandle = NetworkManager.StartCoroutine(ScanInput(NetworkObject));
+            }
+
+            // m_ServerDespawnedOnFirstSpawn prevents the server from always
+            // despawning on the server-side after the first spawn.
+            if (StartDespawned && !m_ServerDespawnedOnFirstSpawn)
+            {
+                m_ServerDespawnedOnFirstSpawn = true;
+                NetworkObject.Despawn(false);
             }
         }
 
         public override void OnNetworkDespawn()
         {
-            if (m_MeshRenderer != null)
-            {
-                m_MeshRenderer.enabled = false;
-            }
+            // It is OK to disable in-scene placed NetworkObjects upon
+            // despawning.  When re-spawned the client-side will re-activate
+            // the GameObject, while the server-side must set the GameObject
+            // active itself.
+            gameObject.SetActive(false);
```

Comment:
Are we sure we are not introducing cases where the game developers see their GameObject activated/deactivated outside their code and get confused/unhappy ?

Reply:
This particular file is only for manual testing purposes.
The NGO SDK side:
- Never disables any GameObject.
- Only enables a GameObject of an in-scene placed NetworkObject on the client side when the server sends the request to spawn it.

The user has to set the GameObject inactive themselves. If they do this, then when user code wants to spawn the same in-scene placed NetworkObject they just set the GameObject on the server-side to active and then spawn the associated NetworkObject.  The client side NetworkObject clones have to be active in order for OnNetworkSpawn to be invoked, which is what we do only for in-scene placed NetworkObjects.

So, a user should never be surprised that the in-scene placed NetworkObject they disabled gets enabled when their code spawns it from the server side.  

Does this make sense?